fog/view/server/src/db_fetcher.rs

```diff
@@ -268,60 +268,34 @@ impl<DB: RecoveryDb + Clone + Send + Sync + 'static> DbFetcherThread<DB> {
         let mut has_more_work = false;
 
         // See whats the next block number we need to load for each invocation we are aware of.
-        let (ingestable_ranges, missing_block_ranges) = {
-            let shared_state = self.shared_state();
-            (
-                shared_state.ingestable_ranges.clone(),
-                shared_state.missing_block_ranges.clone(),
-            )
-        };
+        let ingress_keys = self.shared_state().ingress_keys.clone();
 
-        let next_block_index_per_invocation_id = self.block_tracker.next_blocks(&ingestable_ranges);
+        let next_block_index_per_ingress_key = self.block_tracker.next_blocks(&ingress_keys);
 
         log::trace!(
             self.logger,
             "load_block_data next_blocks: {:?}",
-            next_block_index_per_invocation_id
+            next_block_index_per_ingress_key
         );
 
-        for (ingest_invocation_id, block_index) in next_block_index_per_invocation_id.into_iter() {
-            // Do not attempt to fetch blocks explicitly marked as missing.
-            if BlockTracker::is_missing_block(&missing_block_ranges, block_index) {
```

Comment:
I removed this because I'm worried it can cause a bug

Suppose that an ingest invocation id is chugging along, 0 ... 20

Then, by mistake perhaps, we report 5..10 as a missed block range

Then, the enclave will never load blocks 5..10 for this invocation id, so a bunch of users will have gaps in their RNGs.

Reply:
i think its better to just load these blocks if they are in the database